drivers/sensor/hx711/hx711.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2020 George Gkinis
+ * Copyright (c) 2021 Jan Gnip
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_HX711_H_
+#define ZEPHYR_DRIVERS_SENSOR_HX711_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <device.h>
+#include <kernel.h>
+#include <drivers/sensor.h>
+#include <drivers/gpio.h>
+
+#include <drivers/sensor/hx711.h>
+
+struct hx711_data {
+	const struct device *dout_gpio;
+	const struct device *sck_gpio;
+	const struct device *rate_gpio;
+	struct gpio_callback dout_gpio_cb;
+	struct k_sem dout_sem;
+
+	int32_t reading;
+
+	int offset;
+	struct sensor_value slope;
+	char gain;
+	enum hx711_rate rate;
+	enum hx711_power power;
+};
+
+struct hx711_config {
+	int instance;
+	bool rate_enabled;
+
+	gpio_pin_t dout_pin;
```

Comment:
You should use https://docs.zephyrproject.org/latest/reference/peripherals/gpio.html#c.gpio_dt_spec and appropriate `GPIO_DT_SPEC_INST_GET` macro instead.

Reply:
done.

Reply:
Now you should use `gpio_pin_X_dt` functions instead and get rid of 
```
const struct device *dout_gpio;
const struct device *sck_gpio;
const struct device *rate_gpio;
```
that are not needed anymore. Please see https://docs.zephyrproject.org/latest/reference/peripherals/gpio.html

Reply:
Done.